protocol/storage/miner.go

```diff
@@ -733,23 +739,26 @@ func (sm *Miner) handleQueryDeal(s inet.Stream) {
 // OnNewHeaviestTipSet is a callback called by node, every time the the latest
 // head is updated. It is used to check if we are in a new proving period and
 // need to trigger PoSt submission.
-func (sm *Miner) OnNewHeaviestTipSet(ts types.TipSet) error {
+// If a PoSt computation is started as a result of this new tipset, the returned latch is held until
+// the computation completes.
+func (sm *Miner) OnNewHeaviestTipSet(ts types.TipSet) (*moresync.Latch, error) {
 	ctx := context.Background()
+	doneLatch := moresync.NewLatch(0)
 
 	isBootstrapMinerActor, err := sm.isBootstrapMinerActor(ctx)
 	if err != nil {
-		return errors.Errorf("could not determine if actor created for bootstrapping: %s", err)
+		return doneLatch, errors.Errorf("could not determine if actor created for bootstrapping: %s", err)
 	}
 
 	if isBootstrapMinerActor {
 		// this is not an error condition, so log quietly
 		log.Info("bootstrap miner actor skips PoSt-generation flow")
-		return nil
+		return doneLatch, nil
```

Comment:
Here and below. There is no feedback to the caller that the latch returned here will never be done. It's only used in testing where this isn't a problem, but it could cause problems in the future.

A quick solution would to return a nil latch if no PoSt is started. This method could also return an additional bool parameter to indicate whether or not a PoSt is started. I'm not sure if the latch would work properly if you call done on it before returning, but, if so, that could be done here and at the 3 other non error exit points.

Reply:
The `doneLatch` (with count of zero) is immediately done/open. A call to `Wait` will not block.